examples_tests/61.BlitFilterTest/normalization/normalization.glsl

```diff
@@ -0,0 +1,92 @@
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_INCLUDED_
+#define _NBL_GLSL_BLIT_NORMALIZATION_INCLUDED_
+
+#include <nbl/builtin/glsl/workgroup/arithmetic.glsl>
+
+uint integerDivide_64_32_32(in uint dividendMsb, in uint dividendLsb, in uint divisor)
+{
+	const uint MAX_UINT = ~0u;
+	const float msbRatio = float(dividendMsb) / float(divisor);
+	const uint quotient = uint((msbRatio * MAX_UINT) + msbRatio) + dividendLsb / divisor;
+	return quotient;
+}
+
+#ifndef _NBL_GLSL_BLIT_NORMALIZATION_MAIN_DEFINED_
+
+#ifndef _NBL_GLSL_SCRATCH_SHARED_DEFINED_
+#error "_NBL_GLSL_SCRATCH_SHARED_DEFINED_ must be defined"
+#endif
+
+#define scratchShared _NBL_GLSL_SCRATCH_SHARED_DEFINED_
+
+#include <../normalization/parameters.glsl>
+nbl_glsl_blit_normalization_parameters_t nbl_glsl_blit_normalization_getParameters();
+
+nbl_glsl_blit_normalization_pixel_t nbl_glsl_blit_normalization_getPaddedData(in ivec3 coord);
+void nbl_glsl_blit_normalization_setData(in nbl_glsl_blit_normalization_pixel_t data, in ivec3 coord);
+uint nbl_glsl_blit_normalization_getAlphaHistogramData(in uint index);
+uint nbl_glsl_blit_normalization_getPassedInputPixelCountData();
+
+void nbl_glsl_blit_normalization_main()
+{
+	// Todo(achal): Assert on the CPU that _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_ == WORKGROUP_SIZE
+	uint histogramVal = 0u;
+	if (gl_LocalInvocationIndex < _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_)
+		histogramVal = nbl_glsl_blit_normalization_getAlphaHistogramData(gl_LocalInvocationIndex);
+
+	const uint cumHistogramVal = nbl_glsl_workgroupInclusiveAdd(histogramVal);
+
+	if (gl_LocalInvocationIndex < _NBL_GLSL_BLIT_NORMALIZATION_BIN_COUNT_)
+		scratchShared[gl_LocalInvocationIndex] = cumHistogramVal;
+	barrier();
+
+	const nbl_glsl_blit_normalization_parameters_t params = nbl_glsl_blit_normalization_getParameters();
+
+	const uint outputPixelCount = params.outImageDim.x * params.outImageDim.y * params.outImageDim.z;
```

Comment:
you have some space left over in the paddings of the push constants, store this value there

Reply:
Resolved.